### ISSUE TITLE
"there are no blocks to pop" exception (low priority)

### ISSUE THREAD

**Issue by oxarbitrage** [bitshares/bitshares-core#303]
when the node try to exit clean:

```
483143ms asio       main.cpp:170                  operator()           ] Caught SIGINT attempting to exit cleanly
483143ms th_a       main.cpp:183                  main                 ] Exiting from signal 2
567191ms th_a       db_management.cpp:170         close                ] Database close unexpected exception: {"code":3070001,"name":"pop_empty_chain","message":"there are no blocks to pop","stack":[{"context":{"level":"error","file":"db_block.cpp","line":423,"method":"pop_block","hostname":"","thread_name":"th_a","timestamp":"2017-06-08T09:09:27"},"format":"there are no blocks to pop","data":{}},{"context":{"level":"warn","file":"db_block.cpp","line":431,"method":"pop_block","hostname":"","thread_name":"th_a","timestamp":"2017-06-08T09:09:27"},"format":"","data":{}}]}
```

the node can still exit clean after msg, this issue is to check why the error is happening and check if it can be removed.

related is issue https://github.com/bitshares/bitshares-core/issues/285

**Comment by pmconrad** [comment]
I identified two possible reasons for this:
1. db.close() is called twice - this happens mostly if close() is called and later the enclosing application object is destroyed, because the destructor also calls db.close. So does application.shutdown().
2. db.close is called but the fork/undo database does not contain a sufficient number of blocks. This happens mostly in unit tests.

**Comment by oxarbitrage** [comment]
closing by fix #336
